plugins/out_kinesis_streams/kinesis.c

```diff
@@ -127,18 +127,17 @@ static int cb_kinesis_init(struct flb_output_instance *ins,
      * @param ctx The Kinesis output plugin context.
      */
     flb_plg_debug(ins, "Retrieved port from ins->host.port: %d", ins->host.port);
-    
-    if (ins->host.port >= FLB_KINESIS_MIN_PORT && ins->host.port <= FLB_KINESIS_MAX_PORT) {
-        ctx->port = ins->host.port;
-        flb_plg_debug(ins, "Setting port to: %d", ctx->port);
-    }
-    else if (ins->host.port == 0) {
+
+    if (ins->host.port == 0) {
         ctx->port = FLB_KINESIS_DEFAULT_HTTPS_PORT;
         flb_plg_debug(ins, "Port not set. Using default HTTPS port: %d", ctx->port);
     }
+    else if (ins->host.port == (ctx->port = (uint16_t)ins->host.port)) {
```

Comment:
I'm just wondering if we remove the validation of port: `ins->host.port >= FLB_KINESIS_MIN_PORT && ins->host.port <= FLB_KINESIS_MAX_PORT` is removed here, is there any side effect not to be validated for port number whether inside of the port range or not.

Reply:
The value is casted to `uint16_t`, which falls onto the range [0, 65535] effectively rendering the validation unnecessary since this line also checks for under/over-flow, and 0 has been checked for in line 131.

What sort of side effect do you have in mind?